scripts/base36-uuid.py

```diff
@@ -0,0 +1,216 @@
+        time_mid = (2 ** 16 - 1) & time
+        time >>= 16
+        if time >> 12 != 0:
+            raise ValueError(f'time "{time:#016x}" is too large to fit in')
+        time_hi = (2 ** 12 - 1) & time
+        # sets the version to 1
+        time_hi_version = 1 << 12 | time_hi
+        return (time_low << 32 |
+                time_mid << 16 |
+                time_hi_version)
+
+    @property
+    def msb(self) -> int:
+        return (self.uuid.time_low << 32 |
+                self.uuid.time_mid << 16 |
+                self.uuid.time_hi_version)
+
+    @property
+    def lsb(self) -> int:
+        return (self.uuid.clock_seq_hi_variant << 56 |
+                self.uuid.clock_seq_low << 48 |
+                self.uuid.node)
+
+    @property
+    def timestamp(self) -> (datetime.datetime, int):
+        seconds, decimicro_seconds = divmod(self.uuid.time, DECIMICRO_RATIO)
+        return datetime.datetime.fromtimestamp(seconds), decimicro_seconds
+
+    def print_field(self, field: str, print_in_hex: bool) -> None:
+        def print_num(n: int, bits: int) -> str:
+            if print_in_hex:
+                # each hex char represents 4 bits
+                width = bits // 4
+                print(f'{field} = 0x{n:0{width}x}')
+            else:
+                print(f'{field} = {n}')
+
+        if field == 'lsb':
+            print_num(self.lsb, 64)
+        elif field == 'msb':
+            print_num(self.msb, 64)
+        elif field == 'date':
+            datetime, _ = self.timestamp
+            print(f'date = {datetime}')
+        elif field == 'decimicro_seconds':
+            _, decimicro_seconds = self.timestamp
+            print_num(decimicro_seconds, DECIMICRO_RATIO.bit_length())
+        elif field == 'time':
+            print_num(self.uuid.time, 60)
+        elif field == 'node':
+            print_num(self.uuid.node, 48)
+        else:
+            assert False, f'unknown field: {field}'
+
+    def __str__(self) -> str:
+        return self.encode_with_base36()
+
+
+def test_dencode_base36() -> None:
+    # a minimal pytest
+    uuid = "3fw2_0tj4_46w3k2cpidnirvjy7k"
+    expected_msb = 0x6636ac00da8411ec
+    expected_lsb = 0x9abaf56e1443def0
+    id = TimeUuid.decode_with_base36(uuid)
+    assert id.msb == expected_msb
+    assert id.lsb == expected_lsb
+    assert id.encode_with_base36() == uuid
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Encode and decode timeuuid using base36 representation.")
+    dencode_parser = parser.add_mutually_exclusive_group(required=True)
+    dencode_parser.add_argument('-d', '--decode',
+                                help='Decode base36 encoded timeuuid',
+                                metavar='UUID')
+    dencode_parser.add_argument('-e', '--encode',
+                                nargs=2,
+                                type=lambda x: int(x, 0),
+                                help='encode two hex 64-bit numbers using base36',
```

Comment:
nit: `Encode two 64-bit hex numbers using base36`
(`Decode` starting with a capital letter above and improved gammar)

Reply:
ack.